File: cpp/problems/leetcode/0070_climbing_stairs/solution.cpp
// LeetCode 70
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int climbStairs(int n) {
        if (n <= 2) return n;
        long long a = 1, b = 2;
        for (int i = 3; i <= n; ++i) {
            long long c = a + b;
            a = b;
            b = c;
        }
        return (int)b;
    }
};

int main() {
    assert(Solution().climbStairs(3) == 3);
    assert(Solution().climbStairs(5) == 8);
    cout << "0070 OK" << endl;
    return 0;
}

File: python/problems/leetcode/0070_climbing_stairs/solution.py
"""LeetCode 70: Climbing Stairs."""

from __future__ import annotations


class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2:
            return n
        a, b = 1, 2
        for _ in range(3, n + 1):
            a, b = b, a + b
        return b


if __name__ == "__main__":
    assert Solution().climbStairs(3) == 3
    assert Solution().climbStairs(5) == 8
    print("0070 OK")
